cli/command/container/wait.go

```diff
@@ -28,6 +29,10 @@ func NewWaitCommand(dockerCli *command.DockerCli) *cobra.Command {
 			return runWait(dockerCli, &opts)
 		},
 	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opts.condition, "condition", "not-running", "Wait until a container state reaches the given condition, either 'not-running' (default), 'next-exit', or 'removed'")
```

Comment:
> **not-running**
> It blocks if the container is in a state which is considered to be running, i.e., any of running, paused, or restarting.

From your statement, I would expect that `docker restart` a running container won't make `wait --condition not-running` return, but actually it's not. After reading your codes, I see that `not-running` shares same `waitStop` channel with `next-exit`, the only difference is handling of initial state of container.

I think `not-running` is enough, not sure if there's any scenario for `next-exit`.